store/tikv/region_cache.go

```diff
@@ -831,6 +831,22 @@ func (c *RegionCache) getRegionByIDFromCache(regionID uint64) *Region {
 	return newestRegion
 }
 
+func filterUnavailablePeers(region *pd.Region) {
+	for i := len(region.Meta.Peers) - 1; i >= 0; i-- {
+		peer := region.Meta.Peers[i]
+		available := true
+		for _, downPeer := range region.DownPeers {
+			if peer.Id == downPeer.Id && peer.StoreId == downPeer.StoreId {
+				available = false
+				break
+			}
+		}
+		if !available {
+			region.Meta.Peers = append(region.Meta.Peers[:i], region.Meta.Peers[i+1:]...)
+		}
+	}
```

Comment:
```suggestion
	if len(region.DownPeers) == 0 {
		return
	}
	new := region.Meta.Peers[:0]
	for _, p := range region.Meta.Peers {
		available := true
		for _, downPeer := range region.DownPeers {
			if peer.Id == downPeer.Id && peer.StoreId == downPeer.StoreId {
				available = false
				break
			}
		}
		if available {
			new = append(new, p)
		}
	}
	for i := len(new); i < len(region.Meta.Peers); i++ {
		region.Meta.Peers[i] = nil
	}
	region.Meta.Peers = new
```

What about the standard style of filtering slice?

Reply:
👍